src/views/studio/lib/user-projects-actions.js

```diff
@@ -0,0 +1,54 @@
+import keyMirror from 'keymirror';
+import api from '../../../lib/api';
+import {selectUsername} from '../../../redux/session';
+import {userProjects} from './redux-modules';
+
+const Errors = keyMirror({
+    NETWORK: null,
+    SERVER: null,
+    PERMISSION: null
+});
+
+const Filters = keyMirror({
+    SHARED: null,
+    FAVORITED: null,
+    RECENT: null
+});
+
+const Uris = {
+    [Filters.SHARED]: username => `/users/${username}/projects`,
+    [Filters.FAVORITED]: username => `/users/${username}/favorites`,
+    [Filters.RECENT]: username => `/users/${username}/recent`
+};
+
+const normalizeError = (err, body, res) => {
+    if (err) return Errors.NETWORK;
+    if (res.statusCode === 401 || res.statusCode === 403) return Errors.PERMISSION;
+    if (res.statusCode !== 200) return Errors.SERVER;
+    return null;
+};
+
+const loadUserProjects = type => ((dispatch, getState) => {
+    const state = getState();
+    const username = selectUsername(state);
+    const projectCount = userProjects.selector(state).items.length;
+    const projectsPerPage = 20;
+    dispatch(userProjects.actions.loading());
+    api({
+        uri: Uris[type](username),
+        params: {limit: projectsPerPage, offset: projectCount}
```

Comment:
I'm sure this requires authentication for "recent"? That's how things work currently

Reply:
Yeah it will. We don't have that API route set up yet though, so I'll follow up when it is. (I don't want to submit too much "speculative" code that may not end up actually working).